### ISSUE TITLE
Is CDF5 supposed to be disabled by default in configure?

### ISSUE THREAD

**Issue by edhartnett** [Unidata/netcdf-c#589]
While working on the nc_open() mode flags for issue #392, I noticed that CDF5 is disabled by default:

When I build like this:

`make distclean && autoreconf -i && CPPFLAGS='-I/usr/local/zlib-1.2.11/include -I/usr/local/szip-2.1/include -I/usr/local/hdf5-1.10.1/include' CFLAGS='-g -Wall' LDFLAGS='-L/usr/local/zlib-1.2.11/lib -L/usr/local/szip-2.1/lib -L/usr/local/hdf5-1.10.1/lib' ./configure --enable-logging && make check`

I get:

```
# Features
--------
NetCDF-2 API:		yes
HDF4 Support:		no
NetCDF-4 API:		yes
NC-4 Parallel Support:	no
PNetCDF Support:	no
DAP2 Support:		yes
DAP4 Support:		yes
Diskless Support:	yes
MMap Support:		no
JNA Support:		no
CDF5 Support:		no
```

As seen in the configure.ac file, CDF5 will be off by default. Is this intended?

```
# Check whether we want to enable CDF5 support.
AC_MSG_CHECKING([whether CDF5 support should be enabled (default off)])
AC_ARG_ENABLE([cdf5],
              [AS_HELP_STRING([--enable-cdf5],
                              [build with CDF5 support.])])
test "x$enable_cdf5" = xyes || enable_cdf5=no
AC_MSG_RESULT($enable_cdf5)

```

**Comment by edhartnett** [comment]
Another interesting facet of this is that you can create (but not open) CDF5 files, even with --disable-cdf5. ;-)

And the tests are doing this. This is happening in nc_test/tst_inq_type.c, which creates CDF5 files but never re-opens to check them.

**Comment by WardF** [comment]
Yes, off by default is intended. Thanks for the catch, I’ll fix the tests when I have a chance.

**Comment by edhartnett** [comment]
I am about to put up the test fixes in a PR.

I need these to be fixed to proceed on #392.

In general, tests will have to get a little more sophisticated in handling formats. I have added a function to nc_test4/tst_converts.c:

```
/* Determine how many formats are available, and what they are. */
void
determine_test_formats(int *num_formats, int *format)
{
   int ind = 0;
   int num;

   /* Check inputs. */
   assert(num_formats && format);

   /* We always have classic and 64-bit offset */
   num = 2;
   format[ind++] = NC_FORMAT_CLASSIC;
   format[ind++] = NC_FORMAT_64BIT_OFFSET;

   /* Do we have netCDF-4 and netCDF-4 classic? */
#ifdef USE_NETCDF4
   num += 2;
   format[ind++] = NC_FORMAT_NETCDF4_CLASSIC;
   format[ind++] = NC_FORMAT_NETCDF4;
#endif /* USE_NETCDF4 */

   /* Do we have CDF5? */
#ifdef ENABLE_CDF5
   num++;
   format[ind++] = NC_FORMAT_CDF5;
#endif /* ENABLE_CDF5 */

   *num_formats = num;
}

```

Possibly there are other tests that need this kind of logic (not of the other tests in thie PR needed it, they accomplish this in a different way).

So if I encounter this again I can move this function to a common place and use it in multiple tests.

PR will be up as soon as everything passes my many Jenkins CI builds. ;-)

**Comment by edhartnett** [comment]
BTW I also noticed some cases of NC_NETCDF4 being used outside of ifdefs. This will break once #392 is complete (right now, these are silently creating classic files instead of netCDF-4 files.)

I will take care of those on a different issue and PR, once I encounter them. ;-)
